sources/models/src/lib.rs

```diff
@@ -90,3 +94,174 @@ struct Report {
     name: String,
     description: String,
 }
+
+/// Weak settings are ephemeral and deleted on reboot, regardless of whether or not it
+/// is written by a setting generator.
+#[derive(Deserialize, Serialize, Debug, Clone, Copy, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+#[derive(Default)]
+pub enum Strength {
+    #[default]
+    Strong,
+    Weak,
+}
+
+impl std::fmt::Display for Strength {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Strength::Strong => write!(f, "strong"),
+            Strength::Weak => write!(f, "weak"),
+        }
+    }
+}
+
+derive_fromstr_from_deserialize!(Strength);
+
+/// Struct to hold the setting generator definition containing
+/// command, strength, skip-if-populated
+#[derive(Default, Serialize, std::fmt::Debug, PartialEq)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+pub struct SettingsGenerator {
+    pub command: String,
+    #[serde(default = "Strength::default")]
+    pub strength: Strength,
+    pub skip_if_populated: bool,
+}
```

Comment:
All nits, not blocking:

Since `Strength` default is already `Strong` (it implements `Default`), you don't have to override it with `#[serde`. I think since we want the default value for `skip_if_populated` to be `true`, we could remove the `derive(Default` from this struct and implement it ourselves.

This would also later let you use `..Default::default()` in your custom deserializer, rather than re-listing the default values there.

```suggestion
#[derive(Serialize, std::fmt::Debug, PartialEq)]
#[serde(rename_all = "kebab-case", deny_unknown_fields)]
pub struct SettingsGenerator {
    pub command: String,
    pub strength: Strength,
    pub skip_if_populated: bool,
}
```